dlib/test/dnn.cpp

```diff
@@ -470,6 +470,59 @@ namespace
 
     }
 
+// ----------------------------------------------------------------------------------------
+
+    void test_layer_normalize()
+    {
+        resizable_tensor x(2, 3, 4, 5);
+        resizable_tensor y_cpu(x);
+        tt::tensor_rand rnd(0);
+        rnd.fill_uniform(x);
+        resizable_tensor means(x.num_samples()), invstds(x.num_samples());
+        resizable_tensor gamma(x.num_samples()), beta(x.num_samples());
+        gamma = 1;
+        beta = 0;
+        const float eps = 1e-5;
+        resizable_tensor gradient_input(x), src_grad_cpu(x);
+        resizable_tensor gamma_grad_cpu(x.num_samples()), beta_grad_cpu(x.num_samples());
+        src_grad_cpu = 0;
+        rnd.fill_gaussian(gradient_input);
+        cpu::layer_normalize(eps, y_cpu, means, invstds, x, gamma, beta);
```

Comment:
It looks like `y_cpu` is uninitialized.  Fill it with something like uniform random numbers, so it's got non-zero mean and non-unit variance.

Reply:
Oh wait, nevermind.  `y_cpu` is a pure output.